### PR TITLE
Support removing elements from slice.

### PR DESCRIPTION
**Pull request by fyrchik**
1. Emit error for subslicing of non-byte slices.
2. Support removing slice elements in-place.

### CODE REVIEW

**Review by AnnaShaleva — APPROVED** [review]

File: pkg/interop/util/util.go [hunk 1]
```diff
@@ -17,3 +17,8 @@ func FromAddress(address string) []byte {
 func Equals(a, b interface{}) bool {
 	return false
 }
+
+// Remove removes element with index i from slice.
+// This is done in place and slice must have type `[]byte`.
```
Comment (roman-khimov) [inline pkg/interop/util/util.go:22]: `type []byte`? Really?

**Review by roman-khimov — CHANGES_REQUESTED** [review]

**Comment by roman-khimov** [comment]
It'd be nice to at least mention `util` package in the compiler document.

File: pkg/compiler/codegen.go [hunk 2]
```diff
@@ -517,7 +517,7 @@ func (c *codegen) Visit(node ast.Node) ast.Visitor {
 		return nil
 
 	case *ast.SliceExpr:
-		if isCompoundSlice(c.typeOf(n.X.(*ast.Ident))) {
+		if isCompoundSlice(c.typeOf(n.X.(*ast.Ident)).Underlying()) {
```
[the remark this answers is not in the document]
Comment (fyrchik) [reply pkg/compiler/codegen.go:520]: Fixed.

File: pkg/interop/util/util.go (hunk 1 above)
Comment (fyrchik) [reply pkg/interop/util/util.go:22]: Fixed.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**